### PR TITLE
Custom Hostnames Resource

### PR DESCRIPTION
**Pull request by pcaminog**
New Resource created for support Custom Hostnames in Terraform

This is pending of this PR https://github.com/cloudflare/cloudflare-go/pull/407 to make the update function works.

### CODE REVIEW

File: cloudflare/resource_cloudflare_custom_hostnames.go [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+
+	_, err := client.UpdateCustomHostnameSSL(zoneID, certID, zcso)
+	if err != nil {
+		return errors.Wrap(err, "Error updating Custom Hostname")
+	}
+
+	return resourceCloudflareCustomHostnamesRead(d, meta)
+}
+
+func resourceCloudflareCustomHostnamesRead(d *schema.ResourceData, meta interface{}) error {
```
Comment (jacobbednarz) [inline cloudflare/resource_cloudflare_custom_hostnames.go]: Why aren't the `ssl` option changes reflected in the `Read`? If the remote changes them for any reason, it's not reflected here.

File: cloudflare/resource_cloudflare_custom_hostnames_test.go [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+					resource.TestCheckResourceAttr(
+						resourceName, "zone_id", zoneID),
+					resource.TestCheckResourceAttr(
+						resourceName, "custom_origin_server", "serverone.myinfra.com"),
+					resource.TestCheckResourceAttr(
+						resourceName, "ssl.method", "http"),
+					resource.TestCheckResourceAttr(
+						resourceName, "hostname", "thisis@cool.com"),
+					resource.TestCheckResourceAttr(
+						resourceName, "ssl.custom_hostname_settings.tls13", "on"),
```
Comment (jacobbednarz) [inline cloudflare/resource_cloudflare_custom_hostnames_test.go]: Have you run these integration tests? A [`TypeList`](https://www.terraform.io/docs/extend/schemas/schema-types.html#typelist) is sequentially indice based which means these nested values should have an index in front of them.

**Comment by pcaminog** [comment]
Developing of this PR on hold at the moment due changes on the custom hostname planned for the end of February, to restart next month with the changes needed and adapted to the new model

**Comment by cdloh** [comment]
@pcaminog did those changes end up happening? Is there anything stopping this from going in now?

**Comment by oliverlucasbca** [comment]
Has this been progressed at all?

**Comment by stefanstrydom** [comment]
@pcaminog @jacobbednarz  Has there been any progress on this one by chance?

**Comment by jacobbednarz** [comment]
@pcaminog I've taken a pass at adding support for the current state of affairs in #746. It seems to include a little more than this PR does so would it make sense to close this one off? We can always rejig if the offering changes but this adds a baseline support for anyone waiting on it.

**Comment by jacobbednarz** [comment]
#746 landed so we can close this one and revisit if there are improvements to be made in a separate PR. Thanks!

**Pull request closed without merging**